### DISCUSSION TITLE
Client and server mdx do not match paragraphs

### DISCUSSION THREAD

**Discussion by switz** [mdx-js/mdx#1740] [General]
https://user-images.githubusercontent.com/990166/137380943-eb8afe1e-9de9-44d1-8b62-497a5a7b25fb.mov

See the flicker above as the client side rendering hydrates

MDX:

```
I spent a decade consulting on projects for a variety of companies<Footnote index={1} content={<Box width={240}>
  such as...
</Box>}>hi</Footnote>&nbsp;at every stage, focusing largely on frontend, product, and UX.
```

This is what my code looks like. On the server it renders the footnote as a separate paragraph. On the client, it overtakes it and renders it properly.

Serverside:

<img width="520" alt="image" src="https://user-images.githubusercontent.com/990166/137381126-d30d8d69-ffb1-4ac1-9a85-a356d7b298c3.png">

Clientside:

<img width="662" alt="image" src="https://user-images.githubusercontent.com/990166/137381193-d3ed5bb3-bb8c-490f-b9b9-c1ebe904db81.png">

**Comment by ChristianMurphy** [comment]
@switz what are you using to render MDX server-side? Gatsby? Next.js? Something else?
What version(s) of MDX are you using?
Could you share a runnable example of the issue you are seeing? https://codesandbox.io is a good platform for sharing client/serverside examples

**Reply by switz** [reply]
I'm using nextra/next.js.

Nextra is running on `@mdx-js/loader@^2.0.0-next.9` currently. I can't put a sandbox together this second, but I'll try to make one soon.

**Reply by ChristianMurphy** [reply]
Sounds good, at the same time it may be worth upgrading from the 2.0 pre-release to the stable/generally available release https://github.com/mdx-js/mdx/releases/tag/2.0.0
